### DISCUSSION TITLE
How can I hide track labels programmatically in the LinearGenomeView

### DISCUSSION THREAD

**Discussion by thenath** [GMOD/jbrowse-components#4605] [Q&A] [answered]
Hello,
I would like to know if it is possible and if so how to hide the labels for a track programmatically. I know it's possible via the UI, but I would like this to be the default. 

TIA

**Comment by cmdcolin** [comment]
in your config file you can put

```json
{
  "tracks": [/* your tracks here, just have this here to show you where in your config the configuration part goes */],
  "configuration": {
    "LinearGenomeViewPlugin": {
      "trackLabels": "hidden"
    }
  }
}
```

note that this setting is also saved in localstorage so if the user changes it, it will use the value they chose from localstorage instead of this config. so, tldr, clear your localstorage or use private tab when testing this change

**Comment by thenath** [comment]
Hi Colin,

Thanks so much for coming back to me, I tried adding that to my configuration, so now I have;

```
 const state = jbrowse.createViewState({
                    assembly,
                    tracks,
                    location: this.location,                    
                    configuration : {         
                      LinearGenomeViewPlugin: {
                        "trackLabels": "hidden"
                      },
                      theme... (truncated)
                    }
                  });

                  this.state = state;
                  render(
                    createElement(JBrowseLinearGenomeView, { viewState: state }),
                    document.getElementById('jbrowse_linear_genome_view'),
                  )
```

But no luck, i'm not sure if the theme might be causing a conflict?

**Reply by cmdcolin** [reply, marked as answer]
I can see that this isn't working in embedded also. but, since you are in embedded, you can call programmatic APIs easier (unlike in jbrowse-web which is very configuration oriented)

here is an example that should work for embedded

```
 const state = createViewState({
    assembly,
    tracks,
    location: 'ctgA:1105..1221',
  })
  state.session.view.showTrack('volvox-long-reads-sv-bam')
  state.session.view.setTrackLabels('hidden')


```

this is the particular "API" that I called here: the setTrackLabels function on the "LinearGenomeView" statemodel API https://jbrowse.org/jb2/docs/models/lineargenomeview/#action-settracklabels
